### ISSUE TITLE
Absorbing State Diffusion

### ISSUE THREAD

**Issue by kdmsit** [cloneofsimo/d3pm#2]
Could you please share the code for Absorbing State Forward Diffusion, where all states moved to [MASKED] state eventually?

**Comment by anvuongb** [comment]
I know it's been 2 years but I came across the same problem and this is my implementation for it, just 1 line

```python
  mat = (1-beta)*torch.eye(num_classes) + beta*torch.outer(torch.ones(num_classes), e)
  q_onestep_mats.append(mat)
```
